pandas/tests/io/test_sql.py

```diff
@@ -823,6 +823,19 @@ def test_get_schema_keys(self):
         constraint_sentence = 'CONSTRAINT test_pk PRIMARY KEY ("A", "B")'
         assert constraint_sentence in create_sql
 
+    def test_get_schema_with_index(self):
+        frame = DataFrame({
+            'one': pd.Series([1, 2, 3], index=['a', 'b', 'c']),
+            'two': pd.Series([1, 2, 3], index=['a', 'b', 'c'])
+        })
+        frame.index.name = 'alphabet'
+
+        create_sql = sql.get_schema(frame, 'test', con=self.conn)
+        assert 'alphabet' not in create_sql
+
+        create_sql = sql.get_schema(frame, 'test', con=self.conn, index=True)
+        assert 'alphabet' in create_sql
+
```

Comment:
I think you can use `pytest.parameterize` to individually test the following situations:

* Default argument for `index` (i.e. pass in no value for `index`)
* `index=False`
* `index=True`

Reply:
While I agree with this, I thought it better to stick to the patterns within this module, e.g. https://github.com/pandas-dev/pandas/blob/master/pandas/tests/io/test_sql.py#L814. Perhaps that should be considered when addressing https://github.com/pandas-dev/pandas/issues/22624.

Please advise.

Reply:
I have no problems breaking the pattern here, so I would go ahead and make it more `pytest`-idiomatic in this case.